utils/diffing_with_keys.ml

```diff
@@ -0,0 +1,143 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Florian Angeletti, projet Cambium, Inria Paris             *)
+(*                                                                        *)
+(*   Copyright 2021 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+
+type 'a with_pos = int * 'a
+let with_pos l = List.mapi (fun n x -> n+1,x) l
+let pos (x,_) = x
+let data (_,x) = x
+let mk_pos pos data = pos, data
```

Comment:
At this point, might as well make it a record. :)

Reply:
The record was a bit more noisy outside of the module, but it might be better to have a record rather than the ghost of a record.

Reply:
The record has been restored.